.github/workflows/build_and_release.yml

```diff
@@ -0,0 +1,60 @@
+name: Build and Push Docker Image
+
+on:
+  push:
+    branches:
+      - main
```

Comment:
Look at terraform workflow, how to trigger this to only run on release

Reply:
with proper v* tags

Reply:
For reference:
https://github.com/illumio/terraform-provider-illumio-cloudsecure/blob/0336a5268576a5210edc9a841906952eafb86bff/.github/workflows/release.yml#L13-L16